src/identity/caclient.rs

```diff
@@ -54,14 +63,20 @@ impl CaClient {
         let req = IstioCertificateRequest {
             csr,
             validity_duration: 60 * 60 * 24, // 24 hours
-            metadata: Some(Struct {
-                fields: BTreeMap::from([(
-                    "ImpersonatedIdentity".into(),
-                    prost_types::Value {
-                        kind: Some(Kind::StringValue(id.to_string())),
-                    },
-                )]),
-            }),
+            metadata: {
+                if self.disable_impersonated_identity {
+                    None
```

Comment:
IIRC there is code that checks if the `Result` matches the requested `id` and errors.

Reply:
Yeah, down a few lines at `.verify_san(id)`